### PR TITLE
Extend test suite to test all available files

### PR DESCRIPTION
**Pull request by gaurav**
Instead of maintaining lists of files to be tested within the test suite, I've modified it to automatically identify sets of files that can be tested. This means that new test files can be added to `examples/trees` without modifying the test suite in any way. Lists of files that were tested are reported to Travis-CI (see https://travis-ci.org/gaurav/phylo2owl/builds/193234121#L335).

Each test uses different files as input; for example, if the folder contains an Newick file (with a `.tre` extension) and an OWL file (with an `.owl` extension), then the test suite will automatically run `phylo2owl.py` on the Newick file and compare its output with the OWL file. Similarly, all OWL files in `examples/trees` will be validated using the generic Node shape. Any file-specific SHACL shapes will be tested if the appropriate file (e.g. [`pg_2357.shacl.ttl`](https://github.com/phyloref/phylo2owl/blob/73251b030773202e8a7c3c76b0e2f7fed27b4a55/examples/trees/pg_2357.shacl.ttl)) if present, but won't be tested if that file is missing or misspelled -- in which case the test will be reported as skipped ([see example](https://travis-ci.org/phyloref/phylo2owl/builds/193250223#L346)).

### CODE REVIEW

**Comment by gaurav** [comment]
@hlapp This looks like it's working again -- [and with pretty testing output](https://travis-ci.org/phyloref/phylo2owl/builds/202144673#L337)! Let me know if you think it's ready: meanwhile, [the crowl_et_al experiment](https://github.com/gaurav/phylo2owl/tree/crowl_et_al_experiment) branch is [passing all tests](https://travis-ci.org/gaurav/phylo2owl/builds/202151636#L330) and is ready to go as well! I'll make a separate pull request for that once we close this one.

File: tests/README.md [hunk 1]
```diff
@@ -1 +1,21 @@
...
+This test suite is designed to be used to validate the outputs of the `phylo2owl.py` script. It does this in two ways:
+
+ 1. By comparing an output file with its expected output, and reporting an error if they differ.
+ 2. By validating an output OWL file against a set of SHACL shapes.
+
+You can examine [previous runs of this test suite](https://travis-ci.org/phyloref/phylo2owl) on Travis-CI.
+
+## Test scripts
+
+ - `conftest.py`: A py.test configuration file that sets up the fixtures.
```
Comment (hlapp) [inline tests/README.md:16]: If it's setting up the fixtures, shouldn't it be named `fixtures.py`, or at least `conf_test.py`? At first glance it reads as a test of configuration. Renaming to `fixtures.py` or some such would most easily prevent that mental mistake.

**Review by hlapp — APPROVED** [review]
Looks good to me. Let me know if you want to change the name of the file as per comment, otherwise I'll go ahead and merge. And should we squash this into a single commit, or preserve the whole series?

File: tests/README.md (hunk 1 above)
Comment (gaurav) [reply tests/README.md:16]: That *would* be a better name, but unfortunately `conftest.py` is the hardcoded configuration file name for py.test. On the plus side, anybody familiar with py.test should know what it's for, and it's documented here and [in the file itself](https://github.com/phyloref/phylo2owl/pull/19/files#diff-484462fced51d1a06b1d93b4a44dd535). We could *also* create a `fixtures.py` file containing the fixtures as a py.test module, but [we'd need a `conftest.py` file](https://github.com/pytest-dev/pytest/issues/1569#issuecomment-221790555) to load that plugin, which so we can't really get around it.

**Comment by gaurav** [comment]
I think we should preserve the series: it might be handy to remember what each separate piece does and the fixes necessary to make it all work. I'll remember to recommend one or the other on every pull request!

**Comment by hlapp** [comment]
> That would be a better name, but unfortunately conftest.py is the hardcoded configuration file name for py.test.

I wasn't aware of that, and I think that answers the question as to whether the name is a good idea or not.

**Pull request merged**